crates/storage/src/lazy/lazy_cell.rs

```diff
@@ -109,8 +112,18 @@ where
 {
     fn drop(&mut self) {
         if let Some(key) = self.key() {
-            if let Some(entry) = self.entry() {
-                clear_spread_root_opt::<T, _>(key, || entry.value().into())
+            match self.entry() {
+                Some(entry) => {
+                    clear_spread_root_opt::<T, _>(key, || entry.value().into())
+                }
+                None => {
+                    // the value is not yet in the cache. we need it in there
+                    // though in order to properly clean up.
+                    crate::assert_footprint_threshold!(<T as SpreadLayout>::FOOTPRINT);
+                    let _ = self.get();
+                    let entry = self.entry().expect("entry must exist after prior get");
+                    clear_spread_root_opt::<T, _>(key, || entry.value().into())
```

Comment:
Here, as with other clean up code we can make a differentiation using `SpreadLayout::REQUIRE_DEEP_CLEANING` constant to avoid loading from storage if not needed for clean-up purposes.